bpf/bpf_overlay.c

```diff
@@ -74,6 +74,21 @@ static __always_inline int handle_ipv6(struct __ctx_buff *ctx,
 		 */
 		if (*identity == HOST_ID)
 			return DROP_INVALID_IDENTITY;
+
```

Comment:
The commit message here says:

> Going forward, there are two future changes needed in order:
> * After this commit, any packet associated with the kube-apiserver IP
  must have ID 7 (rather than ID 6; without needing the logic added by
  this commit).
> * Once v1.12.y is released, we can remove this specific code as we can
  expect all of this traffic to be transmitted with "kube-apiserver"
  identity (7).

I think what you're saying is that we need to make subsequent changes to the code to achieve each of these steps. But the tense ("After this commit, any packet ... must have ID 7") makes it ambiguous whether you're talking about the code now or some future version. Note that this commit message will end up backported to 1.11.y as well, so this could confuse future readers if they're investigating this code in v1.11 during an upgrade in future.

Reply:
Updated commit msg, hopefully it's clearer now.